### DISCUSSION TITLE
Deployment guide: Digital Ocean

### DISCUSSION THREAD

**Discussion by michaelbromley** [vendurehq/vendure#1535] [Deployment]
Collecting information / tips / gotchas about deploying Vendure to Digital Ocean. The idea is to consolidate into an official guide.

**Comment by ramkishore-yerubandi** [comment]
I deployed Vendure as an App in DigitalOceans without any issue. Used pm2 to deploy both server and worker. Here is the command in my package.json

 "pm-start": "pm2-runtime process.json",

process.json:
{
  "apps": [
    {
      "script": "./dist/index.js",
      "name": "service"
    },
    {
      "script": "./dist/index-worker.js",
      "name": "worker"
    }
  ]
}

**Comment by jnahlovsky** [comment]
As a starter or demo tryout (what I actually did)...

If you rather prefer Docker (container) solution, you can follow this guide https://github.com/vendure-ecommerce/vendure-docker-compose
and simply run it on Ubuntu with preinstalled Docker engine (choose it while creating droplet), then just git clone the repo somewhere on the server and run "docker compose up" from there... it just takes some time to build it, if you're on the budget and have only one shared CPU.

The only downside in my eyes is that it could have been done fully prepared, already built, and published to Dockerhub (under an official Vendure publisher) and it would be only about downloading it and running the apps immediately. It would be a much quicker and more convenient solution.
You can of course build images locally on your machine and publish it yourself though.
